### DISCUSSION TITLE
Old scripts

### DISCUSSION THREAD

**Discussion by ghost** [ldtteam/minecolonies#10325] [Q&A] [answered]
Hello, I wanted to redo something into the code, but I ran into the problem that many scripts are already outdated, and if you download the file from the mod itself and then decompile it, you may encounter the problem of file Obfuscation. Please update the scripts on GitHub

**Comment by MotionlessTrain** [comment]
Which scripts are you referring to?
The source code on GitHub is up to date, as it is used to compile and build the mod

**Reply by ghost** [reply]
Yes, the source code of the mod and many of the scripts there are already outdated, please update them

**Reply by Talyda** [reply]
You might need to specify what you are actually talking about

**Reply by ghost** [reply]
I'm talking about the script AbstractBuildingGuards.class the old code is written in it, the code in the new versions has already been updated

**Reply by MotionlessTrain** [reply, marked as answer]
There is no AbstractBuildingGuards.class on github, though. You would need to decompile the mod itself (as the [readme](https://github.com/ldtteam/minecolonies/tree/version/main) specifies), and then it compiles the src code (which reflects the most recent version, at time of forking) into those class files
